### PR TITLE
DIRS Rules round 2

### PR DESCRIPTION
**Pull request by steve-dlp**
new rule

### CODE REVIEW

File: ValidationWeb/Content/SqlRules/DIRSRuleSet/dirs-rule-57.03.0009.sql [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+MISSING_OR_INVALID_DisciplineAction	 At least one DisciplineAction is required per studentDisicplineIncidentBehaviorAssociation  
+*/
+DECLARE @RuleId VARCHAR(32) = '57.03.0009';
+DECLARE @Message NVARCHAR(MAX) = 'At least one DisciplineAction is required per studentDisicplineIncidentBehaviorAssociation';
+DECLARE @IsError BIT = 1;
+
+with failed_rows as ( 
+SELECT distinct 
+	sdiba.Schoolid,
+  sdiba.StudentUniqueId,
```
Comment (ckimdbl) [inline ValidationWeb/Content/SqlRules/DIRSRuleSet/dirs-rule-57.03.0009.sql:11]: needs to join to edfi.Student to get StudentUniqueId

File: ValidationWeb/Content/SqlRules/DIRSRuleSet/dirs-rule-57.05.0002.sql [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+  [mn].[DisciplineIncidentAnonymousNonOffender] DIANO
+INNER JOIN
+  [edfi].School ON School.SchoolId = DIANO.SchoolId
+INNER JOIN 
+  [edfi].Descriptor NonOffenderType ON NonOffenderType.DescriptorId = DIANO.NonOffenderTypeDescriptorId
+WHERE 
+  NonOffenderType.ShortDescription IN ('Teacher', 'District Employee')
+	AND
+  (DIANO.NonOffenderCount is NULL or DIANO.NonOffenderCount < 1)
+  )
```
Comment (ckimdbl) [inline ValidationWeb/Content/SqlRules/DIRSRuleSet/dirs-rule-57.05.0002.sql:32]: extra )

File: ValidationWeb/Content/SqlRules/DIRSRuleSet/dirs-rule-57.05.0003.sql [hunk 3]
```diff
@@ -0,0 +1,38 @@
+/*
+MISSING_NONOFFENDERENROLLEDCOUNT
+NonOffenderEnrolledCount is required if NonOffenderType - 4 ('MARSS information unknown') and NonOffenderNotEnrolledCount = 0
+*/
+
+
+DECLARE @RuleId VARCHAR(32) = '57.05.0003';
+DECLARE @Message NVARCHAR(MAX) = 'EstimatedStudentsEnrolled is required if NonOffender Type  =  4 MARSS information unknown and EstimatedStudentsNotEnrolled = 0 ;
```
Comment (ckimdbl) [inline ValidationWeb/Content/SqlRules/DIRSRuleSet/dirs-rule-57.05.0003.sql:8]: missing closing '

File: ValidationWeb/Content/SqlRules/DIRSRuleSet/dirs-rule-57.05.0004.sql [hunk 4]
```diff
@@ -0,0 +1,35 @@
+/*
+MISSING_NONOFFENDERNOTENROLLEDCOUNT	
+NonOffenderNotEnrolledCount is required if NonOffenderType - 4 ('MARSS information unknown') and NonOffenderEnrolledCount = 0
+*/
+
+
+DECLARE @RuleId VARCHAR(32) = '57.05.0004';
+DECLARE @Message NVARCHAR(MAX) = 'EstimatedStudentsNotEnrolled is required if NonOffender Type  =  4 MARSS information unknown and EstimatedStudentsEnrolled = 0 ;
```
Comment (ckimdbl) [inline ValidationWeb/Content/SqlRules/DIRSRuleSet/dirs-rule-57.05.0004.sql:8]: missing closing '

File: ValidationWeb/Content/SqlRules/DIRSRuleSet/dirs-rule-57.05.0007.sql [hunk 5]
```diff
@@ -0,0 +1,33 @@
...
+
+WHERE 
+
+  DIANO.DidInjuryOccur = 1
+  AND DIANO.DidSeriousBodilyInjuryOccur is null
+  AND School.LocalEducationAgencyId = @districtid)
+  
+INSERT INTO 
+rules.RuleValidationDetail (RuleValidationId, Id, RuleId, IsError, [Message])
+SELECT	@RuleValidationId, StudentUniqueId, @RuleId RuleId, @IsError IsError, @Message  + ' , IncidentIdentifier: ' + cast(IncidentIdentifier as varchar) + ', DidInjuryOccur: ' + cast(DidInjuryOccur as varchar)[Message]
```
Comment (ckimdbl) [inline ValidationWeb/Content/SqlRules/DIRSRuleSet/dirs-rule-57.05.0007.sql:32]: StudentUniqueId not included in the failed_rows CTE

**Review by ckimdbl — CHANGES_REQUESTED** [review]
please refer to comments I've left for changes.

File: ValidationWeb/Content/SqlRules/DIRSRuleSet/dirs-rule-57.03.0009.sql (hunk 1 above)
Comment (steve-dlp) [reply ValidationWeb/Content/SqlRules/DIRSRuleSet/dirs-rule-57.03.0009.sql]: Updated

File: ValidationWeb/Content/SqlRules/DIRSRuleSet/dirs-rule-57.05.0002.sql (hunk 2 above)
Comment (steve-dlp) [reply ValidationWeb/Content/SqlRules/DIRSRuleSet/dirs-rule-57.05.0002.sql]: updated

File: ValidationWeb/Content/SqlRules/DIRSRuleSet/dirs-rule-57.05.0004.sql (hunk 4 above)
Comment (steve-dlp) [reply ValidationWeb/Content/SqlRules/DIRSRuleSet/dirs-rule-57.05.0004.sql]: updated

File: ValidationWeb/Content/SqlRules/DIRSRuleSet/dirs-rule-57.05.0007.sql (hunk 5 above)
Comment (steve-dlp) [reply ValidationWeb/Content/SqlRules/DIRSRuleSet/dirs-rule-57.05.0007.sql]: Updated to use Schoolid since anonymous offender has no studentid

**Pull request merged**